#!/bin/sh -e
## 50_table-add-to-text-mode-hook.dpatch by Peter S Galbraith <psg@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add table-add-to-text-mode-hook.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/emacs-goodies-el/table.el emacs-goodies-el/elisp/emacs-goodies-el/table.el
--- /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/emacs-goodies-el/table.el	2003-10-08 20:00:03.000000000 -0400
+++ emacs-goodies-el/elisp/emacs-goodies-el/table.el	2003-10-09 22:15:32.000000000 -0400
@@ -799,11 +799,26 @@
 See `table-insert' for examples about how to use."
   :tag "Table"
   :prefix "table-"
+  :link '(emacs-commentary-link "table.el")
+  :link '(url-link "http://table.sourceforge.net/")
   :group 'editing
   :group 'wp
   :group 'paragraphs
   :group 'fill)
 
+(defcustom table-add-to-text-mode-hook nil
+  "If non-nil, add automatic table detection to `text-mode-hook'.
+Tables in visited files will be automatically editable using the `table'
+package.  Alternatively, use `table-recognize' manually in buffers."
+  :type 'boolean
+  :group 'table
+  :require 'table
+  :set (lambda (symbol value)
+         (set-default symbol value)
+         (if value
+             (add-hook 'text-mode-hook 'table-recognize)
+           (remove-hook 'text-mode-hook 'table-recognize))))
+
 (defcustom table-time-before-update 0.2
   "*Time in seconds before updating the cell contents after typing.
 Updating the cell contents on the screen takes place only after this
